src/classes/BDI_DataImportService.cls

```diff
@@ -704,62 +704,119 @@ global with sharing class BDI_DataImportService {
+                parametersByName.put('dataImportField', dataImportField);
+                mapFieldsForDIObjectCore(parametersByName, dataImportFieldToTargetField);
+            }
+        }
 
-                    // make sure they have permissions to modify the target field
-                    // we can't do this in all tests, because many profiles won't have FLS set for all our new fields!
-                    if (!Test.isRunningTest() || failFieldLevelSecurity) {
-                        fieldDescribe = UTIL_Describe.getFieldDescribe(targetObjectName, fieldName);
-                        Set<String> setAuditFields = new Set<String>{
-                            'createddate', 'createdbyid', 'lastmodifieddate', 'lastmodifiedbyid'
-                        };
-                        Boolean isAuditField = setAuditFields.contains(fieldName.toLowerCase());
-
-                        if (failFieldLevelSecurity ||
-                            fieldDescribe == null ||
-                            !fieldDescribe.isCreateable() ||
-                            (!isAuditField && !fieldDescribe.isUpdateable())) {
-                                throw (new FLSException(
-                                    String.format(System.Label.flsError, new List<String>{ targetObjectName + '.' + fieldName }
-                                )));
-                        }
+        return dataImportFieldToTargetField;
+    }
+
+    /*******************************************************************************************************
+    * @description The core logic for the method mapFieldsForDIObject
+    * @param parametersByName a Map containing parameters for this function. The key being the parameter name and the value being the actual parameter.
+    * @param dataImportFieldToTargetField Map<String, String> a map of DataImport fields to underlying object fields (eg, Donation_Amount__c => Amount)
+    * @return void.
+    */
+    private static void mapFieldsForDIObjectCore(Map<String, String> parametersByName, Map<String, String> dataImportFieldToTargetField) {
+        String targetFieldName = parametersByName.get('targetFieldName');
+        String dataImportObjectName = parametersByName.get('dataImportObjectName');
+        String targetObjectName = parametersByName.get('targetObjectName');
+        String dataImportField = parametersByName.get('dataImportField');
+        if (targetFieldName != null && targetFieldName != '') {
+            List<String> parts = targetFieldName.split('\\.', 2);
+            if (parts.size() == 2 && parts[0] == dataImportObjectName && parts[1] != '') {
+                String fieldName = parts[1];
+
+                //if we aren't in the npsp namespace, remove those tokens from help text
+                //also removes tokens from actual detected namespace, if we happen to be
+                //in a non-npsp namespace
+                if (UTIL_Namespace.getNamespace() != 'npsp') {
```

Comment:
I think this would only be true if we're in a non-namespaced org, in which case if `String.isBlank(UTIL_Namespace.getNamespace())` would be clearer. It's a bit confusing because of the way UTIL_Namespace.getNamespace() works, unless we include that class in every package we develop? @force2b Do you have any insight here?

Reply:
I like the idea, will do this change and test.